[camelcase] False positive when using leading underscore when destructuring with computed object property name.

**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.16.1
* **npm Version:** v6.14.2

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
{
    "parserOptions": {
        "ecmaVersion": 2020
    },
    "rules": {
        "camelcase": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
const obj = {};

const { ['foo']: _foo } = obj;
```

<!-- Paste the command you used to run ESLint: -->
```bash
eslint ./main.js
```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**

Error:

```
<redacted>\main.js
  4:18  error  Identifier '_foo' is not in camel case  camelcase

✖ 1 problem (1 error, 0 warnings)
```

Seems to be caused by the computed object property name in the `{ ['foo']: _foo }` destructuring syntax. There is no error if it is changed to `{ foo: _foo }`.

**Are you willing to submit a pull request to fix this bug?**

Not at this time.

--- a/lib/rules/camelcase.js
+++ b/lib/rules/camelcase.js
@@ -186,7 +186,7 @@ module.exports = {
 
                         const assignmentKeyEqualsValue = node.parent.key.name === node.parent.value.name;
 
-                        if (isUnderscored(name) && node.parent.computed) {
+                        if (nameIsUnderscored && node.parent.computed) {
                             report(node);
                         }
 

--- a/tests/lib/rules/camelcase.js
+++ b/tests/lib/rules/camelcase.js
@@ -92,6 +92,14 @@ ruleTester.run("camelcase", rule, {
             options: [{ properties: "never" }]
         },
         {
+            code: "const { ['foo']: _foo } = obj;",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "const { [_foo_]: foo } = obj;",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
             code: "var { category_id } = query;",
             options: [{ ignoreDestructuring: true }],
             parserOptions: { ecmaVersion: 6 }

